content/en/docs/performance/resource_consumption.md

```diff
@@ -0,0 +1,122 @@
+## 3.1 Scenario where the POD CPU limit is not set, test the Kmesh eBPF CPU limit.
+![resource_test3](images/resource_test3.png)
+
+Create 8 instances of APP A, set the CPU limit to unlimited. Gradually modify the number of processes generating load for APP A until the node's CPU usage reaches 100%, collect the CPU usage of Kmesh eBPF.
+
+Test results (8-core CPU totaling 8000m):
+|threads|APP A CPU Usage|eBPF CPU usage|
+|--|--|--|
+|100|12.3%|1%|
+|500|35%|4.1%|
+|1000|61.7%|8.8%|
+|3000|67%|9.5%|
+
+The total CPU of the system is 8 cores, which is equivalent to 8000m. When there are 3000 concurrent processes, the CPU of the node reaches 100% utilization. At this time, APP A consumes 67% of the CPU, and Kmesh eBPF consumes approximately 9.5%.
+
+**Conclusion：**
+
+- **APP A consumes much more CPU than eBPF, so it is not possible to overload eBPF. In a scenario tested with Fortio, Kmesh eBPF consumes a maximum of 9.5% CPU.**
+- **Further testing is needed to determine the maximum CPU consumption limit for eBPF itself.**
+
+## 3.2 eBPF CPU stress test, by adding infinite loops/large for loops in the eBPF code to boost CPU usage.
+The [eBPF official documentation](https://ebpf-docs.dylanreimerink.nl/linux/concepts/verifier/) points out that eBPF programs have a robust security mechanism that automatically detects infinite loops and strictly limits the number of iterations in for loops. In the current Kernel version (v5.10), eBPF programs support a maximum of 65535 iterations in for loops.
+
+Therefore, add 65535 for loops in the code and test it.
+```c
+int i;
+for (i=0;i<65535;i++) {
+    bpf_printk("increase cpu usage");
+}
+```
+
+Testing result:
+![resource_result4](images/resource_test_result4.png)
+
+When the CPU of the node is running at 100%, Kmesh eBPF consumes approximately 99.3% of the CPU. This stress test lasted for 10 minutes, during which the kernel and services within the cluster continued to run stably.
+
+**Conclusion: In the Kmesh eBPF component, when adding support for a maximum number of for loop scenarios, eBPF can consume all CPU resources. However, the kernel's security mechanisms ensure the stable operation of the system.**
+
+
+# 4. Kmesh eBPF Memory limit test
+The memory consumption of eBPF has an upper limit: as stated in the [official documentation](https://ebpf-docs.dylanreimerink.nl/linux/concepts/resource-limit/), this limit is set through the `memory.max` setting in cGroup configurations.
+
+However, based on the current implementation of Kmesh, memory is allocated at the start of Kmesh, and does not increase during runtime. To verify the memory usage, the following test was conducted.
+
+## 4.1 Create 1, 100, and 1000 services respectively in the cluster, and record the eBPF memory consumption.
+Testing result:
+|service number|eBPF Memory usage|
+|--|--|
+|1|23m|
+|100|23m|
+|1000|23m|
+
+**Test results: The eBPF memory consumption of Kmesh is 23MB and remains constant, regardless of the number of services.**
```

Comment:
Are these services associated with real pods? Won't the ebpf map grow as the number of servcies grow?

Reply:
Those are mock services, but I can see the data in the bpf Map. no change in the size of eBPF map